x/bor/keeper/side_msg_server.go

```diff
@@ -344,6 +391,181 @@ func (s sideMsgServer) PostHandleMsgBackfillSpans(ctx sdk.Context, msgI sdk.Msg,
+			return err
+		}
+
+		// Inclusive overlap: [reqStart, reqEnd] overlaps [dtStart, dtEnd]
+		reqStart, reqEnd := msg.DowntimeRange.StartBlock, msg.DowntimeRange.EndBlock
+		if reqStart <= downtime.EndBlock && reqEnd >= downtime.StartBlock {
+			overlapCount++
+		}
+	}
+
+	otherProducers := len(producers) - 1
+	// Reject only if EVERY other producer has a planned downtime AND all of them overlap the requested range
+	if otherProducers > 0 && foundCount == otherProducers && overlapCount == otherProducers {
+		return fmt.Errorf("producer with id %d has overlapping planned downtime with all other producers", validatorId)
+	}
+
+	if err := s.k.ProducerPlannedDowntime.Set(ctx, validatorId, types.BlockRange{
+		StartBlock: msg.DowntimeRange.StartBlock,
+		EndBlock:   msg.DowntimeRange.EndBlock,
+	}); err != nil {
+		return err
+	}
+
+	params, err := s.k.GetParams(ctx)
+	if err != nil {
+		logger.Error("error fetching params", "error", err)
+		return err
+	}
+
+	lastSpan, err := s.k.GetLastSpan(ctx)
+	if err != nil {
+		logger.Error("error fetching last span", "error", err)
+		return err
+	}
+
+	latestActiveProducer, err := s.k.GetLatestActiveProducer(ctx)
+	if err != nil {
+		logger.Error("Error occurred while getting latest active producer", "error", err)
+		return err
+	}
+
+	latestFailedProducer, err := s.k.GetLatestFailedProducer(ctx)
+	if err != nil {
+		logger.Error("Error occurred while getting latest failed producer", "error", err)
+		return err
+	}
+
+	for producerID := range latestFailedProducer {
+		delete(latestActiveProducer, producerID)
+	}
+
+	delete(latestActiveProducer, validatorId)
+
+	hasOverlappingSpan := func() (bool, error) {
+		// Walk backwards from the last span and return the smallest ID that still overlaps [dtStart, dtEnd].
+		cur := lastSpan
+
+		for {
+			isDown, derr := s.k.IsProducerDownForBlockRange(ctx, cur.StartBlock, cur.EndBlock, validatorId)
+			if derr != nil {
+				logger.Error("failed to check producer downtime overlap", "spanId", cur.Id, "error", derr)
+				return false, derr
+			}
+
+			if isDown {
+				return true, nil
+			}
+
+			// Stop once we're past the downtime start window (older spans won't overlap)
+			if cur.Id == 0 || cur.EndBlock < msg.DowntimeRange.StartBlock {
+				break
+			}
+
+			// Move to previous span; if not found, stop scanning.
+			prev, gerr := s.k.GetSpan(ctx, cur.Id-1)
+			if gerr != nil {
+				logger.Debug("stopping backward scan while traversing previous spans",
+					"fromSpanId", cur.Id, "error", gerr)
+				break
+			}
+
+			cur = prev
+		}
+
+		return false, nil
+	}
+
+	foundOverlappingSpan, err := hasOverlappingSpan()
+	if err != nil {
+		return err
+	}
+
+	if !foundOverlappingSpan {
+		// No spans overlap the downtime window; nothing to do.
+		return nil
+	}
+
+	if err := s.k.AddNewVeblopSpan(ctx, validatorId, msg.DowntimeRange.StartBlock, msg.DowntimeRange.StartBlock+params.SpanDuration, lastSpan.BorChainId, latestActiveProducer, uint64(ctx.BlockHeight())); err != nil {
```

Comment:
Would it be better to set the end block of the new span as the block right before the next span planned previously? It would have less impact on the following span.